syz-fuzzer/fuzzer.go

```diff
@@ -453,13 +455,17 @@ func (fuzzer *Fuzzer) poll(needCandidates bool, stats map[string]uint64) bool {
 }
 
 func (fuzzer *Fuzzer) sendInputToManager(inp rpctype.Input) {
-	a := &rpctype.NewInputArgs{
-		Name:  fuzzer.name,
-		Input: inp,
-	}
-	if err := fuzzer.manager.Call("Manager.NewInput", a, nil); err != nil {
-		log.SyzFatalf("Manager.NewInput call failed: %v", err)
-	}
+	fuzzer.newInputSem.Acquire(context.Background(), 1)
```

Comment:
Does it really happen that we block here?
Potentially it can be speed up with batching. All procs just send their inputs to a large channel. There is 1 goroutine that reads everything that there and sends a batch, then waits for the next inputs.

Reply:
> Does it really happen that we block here?

It may happen theoretically, and the prevention is quite cheap, so I think it's worth it.

> Potentially it can be speed up with batching. 

I've been thinking about this, but when I looked into the `syz-manager/rpc.go`, I did not immediately see why it would run faster with batching, all processing must really happen one-by-one. RPC calls are also sent over the same socket asynchronously, so de facto there's already a list of requests to process.

Reply:
I suspect lots of the overhead is per-call.